### PR TITLE
Updated/Standardized navbar of member-site  

### LINKED ISSUE #261: Standardize Navbar across Real Dev Squad
### Design of New Navbar:
https://www.figma.com/file/bpFHG9hXM13GSLmTkoNsrF/Navbar?node-id=0%3A1
### For CSS, refer to the link below
https://github.com/Real-Dev-Squad/website-www/blob/develop/css/navbar.css

## Few Details 
- font : Roboto
- weight: 700
- active color : #49A82E
- hover color : #49A82E
- font size : 16 px (try making it in rem)
- active underline height : 3px (color : #49A82E)
- RDS Logo Size - 50x50

### Pay Attention
=> Underline should be under the text as shown in the image and not inside the container in which the text persists.
=> Underline should be only under the active tab, it should not be visible on hover.
=> The full navbar should be changed to the toggle menu when the screen width is less than 970px.
=> User should be directed to `my.realdevsquad.com` on clicking `User Greet`

### PR DESCRIPTION
**Pull request by Kshashank99**
I have made the changes as per the instructions. Please review my PR.
Updated Navbar :
image
image

Closes #261

### CODE REVIEW

File: src/components/UI/navbar/navbar.module.scss [hunk 1]
```diff
@@ -4,24 +4,34 @@
...
+  display: flex;
+  align-items: center;
+  height: 5.25rem;
 }
 .infoNavbar nav li {
   float: left;
 }
 .infoNavbar nav li a {
+  font: Roboto;
+  font-size: 20px;
```
Comment (sumitd94) [inline src/components/UI/navbar/navbar.module.scss:16]: rem

File: src/components/UI/navbar/navbar.module.scss [hunk 2]
```diff
@@ -4,24 +4,34 @@
...
   display: block;
-  border-radius: 0.5rem;
   color: #ffffff;
-  font-weight: bolder;
+  font-weight: 700;
   text-align: center;
-  padding: 14px 16px;
+  padding: 4px 0px;
+  margin: 0 16px;
```
Comment (sumitd94) [inline src/components/UI/navbar/navbar.module.scss:22]: rem

File: src/components/UI/side-drawer/index.js [hunk 3]
```diff
@@ -22,7 +23,16 @@ const SideDrawer = ({ open, close }) => {
             >
               &#9747;
             </span>
-            Real Dev Squad
+
+            {/* <div>Real Dev Squad</div> */}
+            <a className={classNames.logo} href="https://realdevsquad.com">
```
[the remark this answers is not in the document]
Comment (sumitd94) [reply src/components/UI/side-drawer/index.js:26]: Can you please move the url to a constants file and then use it here?

File: src/components/UI/side-drawer/side-drawer.module.scss [hunk 4]
```diff
@@ -16,7 +16,13 @@
   text-align: center;
   padding-top: 20px;
   color: #1d1283;
+  display: flex;
+  flex-direction: row-reverse;
 }
+.logo {
+  margin-right: 45px;
```
Comment (sumitd94) [inline src/components/UI/side-drawer/side-drawer.module.scss:23]: rem please

File: src/components/UI/side-drawer/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Kshashank99) [reply src/components/UI/side-drawer/index.js]: Hey @sumitd94 changes are done. Please review it

File: src/components/UI/navbar/index.js [hunk 5]
```diff
@@ -18,7 +19,14 @@ const Navbar = ({ drawerToggleClicked }) => {
         </div>
         <ul>
           <li>
-            <a href="https://www.realdevsquad.com/">Home</a>
+            <a href="https://realdevsquad.com">
+              <Image
+                width="45px"
+                height="45px"
+                src="/images/Real-Dev-Squad@1x.png"
+                alt="real-dev squad"
```
[the remark this answers is not in the document]
Comment (Kshashank99) [reply src/components/UI/navbar/index.js:35]: sorry @harshith-venkatesh . I didn't get what gap are you talking about. Please clarify.

File: src/components/UI/navbar/navbar.module.scss [hunk 6]
```diff
@@ -4,24 +4,34 @@
...
+  height: 5.25rem;
 }
 .infoNavbar nav li {
   float: left;
 }
 .infoNavbar nav li a {
+  font: Roboto;
+  font-size: 20px;
   display: block;
-  border-radius: 0.5rem;
```
[the remark this answers is not in the document]
Comment (Kshashank99) [reply src/components/UI/navbar/navbar.module.scss:13]: border radius was not present in the new design suggested by rahil and rohan.

File: src/components/UI/navbar/navbar.module.scss (hunk 1 above)
Comment (Kshashank99) [reply src/components/UI/navbar/navbar.module.scss]: done

File: src/components/UI/navbar/navbar.module.scss (hunk 2 above)
Comment (Kshashank99) [reply src/components/UI/navbar/navbar.module.scss]: done

File: src/components/UI/navbar/index.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (harshith-venkatesh) [reply src/components/UI/navbar/index.js:35]: `alt="real-dev squad"`
we usually go with `real-dev-squad` or `realDevSquad` or in this case `real dev squad` 
Any reason there is a difference here

[the remark this answers is not in the document]
Comment (Kshashank99) [reply src/components/UI/navbar/index.js]: changes committed. Please check.

File: src/components/UI/side-drawer/side-drawer.module.scss (hunk 4 above)
Comment (Kshashank99) [reply src/components/UI/side-drawer/side-drawer.module.scss]: Done.

File: src/components/UI/navbar/index.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (swarajpure) [reply src/components/UI/navbar/index.js]: Why alt text is in kebab or camel case @harshith-venkatesh?
Should it not be in simple english? If a non-technical person sees a broken image, alt text is what is shown to them, right? For them to understand, does simple english `Real Dev Squad Logo` not make more sense?
cc @Kshashank99

File: src/constants/AppConstants.js [hunk 7]
```diff
@@ -1,3 +1,8 @@
 export const SET_MEMBERS = 'SET_MEMBERS';
 export const SET_ERRORS = 'SET_ERRORS';
 export const HOST_NAME = 'members.realdevsquad.com';
+export const WELCOME_LINK = 'https://welcome.realdevsquad.com/';
+export const EVENTS_LINK = 'https://www.realdevsquad.com/events.html';
+export const MEMBERS_LINK = 'https://members.realdevsquad.com/';
+export const CRYPTO_LINK = 'https://crypto.realdevsquad.com/';
+export const STATUS_LINK = 'https://status.realdevsquad.com/';
```
Comment (swarajpure) [inline src/constants/AppConstants.js:8]: Perfect 🔥

File: src/components/UI/navbar/index.js [hunk 8]
```diff
@@ -18,24 +27,31 @@ const Navbar = ({ drawerToggleClicked }) => {
...
           <li>
-            <a href="https://www.realdevsquad.com/events.html">Events</a>
+            <a href={EVENTS_LINK}>Events</a>
           </li>
           <li>
-            <a
-              href="https://members.realdevsquad.com/"
-              style={{ color: '#87D870' }}
-            >
+            <a href={MEMBERS_LINK} style={{ color: '#87D870' }}>
```
Comment (swarajpure) [inline src/components/UI/navbar/index.js:46]: Did we try giving a className `active` or something instead of CSS in JSX?

File: src/components/UI/navbar/navbar.module.scss [hunk 9]
```diff
@@ -4,24 +4,34 @@
...
+  font: Roboto;
+  font-size: 1.25rem;
   display: block;
-  border-radius: 0.5rem;
   color: #ffffff;
-  font-weight: bolder;
+  font-weight: 700;
   text-align: center;
-  padding: 14px 16px;
+  padding: 	0.25rem 0;
```
Comment (swarajpure) [inline src/components/UI/navbar/navbar.module.scss:21]: Please fix the formatting here

**Review by harshith-venkatesh — APPROVED** [review]
Good work👍

File: src/components/UI/navbar/navbar.module.scss (hunk 9 above)
Comment (harshith-venkatesh) [reply src/components/UI/navbar/navbar.module.scss:21]: I was in the thought it might be my mobile issue, there is indeed a formatting issue I see

File: src/components/UI/navbar/index.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (harshith-venkatesh) [reply src/components/UI/navbar/index.js]: I had commented in the same thought @swarajpure , we can go in any case and simple text is definitely preferred here , the alt over there was real-dev squad , that's the reason I suggested to go in one way as I assumed there is no - in real dev

[the remark this answers is not in the document]
Comment (harshith-venkatesh) [reply src/components/UI/navbar/index.js]: I also vouch for `Real Dev Squad Logo` here
Cc: @Kshashank99

[the remark this answers is not in the document]
Comment (swarajpure) [reply src/components/UI/navbar/index.js]: Cool!

File: src/components/UI/navbar/index.js (hunk 8 above)
Comment (Kshashank99) [reply src/components/UI/navbar/index.js:46]: <a className="active" href={MEMBERS_LINK} >Members</a> .  Like this?

Comment (Kshashank99) [reply src/components/UI/navbar/index.js:46]: i guess it was already in inline css before i changed anything. If you feel that it require a separate class for that css then let me know?

File: src/components/UI/navbar/navbar.module.scss (hunk 9 above)
Comment (Kshashank99) [reply src/components/UI/navbar/navbar.module.scss]: Formatted the code. please review

**Comment by Kshashank99** [comment]
Hey @rohan09-raj, @sumitd94  and @harshith-venkatesh Please review the PR.
Before 
image
image

After
image
image

File: package.json [hunk 10]
```diff
@@ -1,60 +1,64 @@
...
-    "eslint-plugin-react-hooks": "^4.2.0",
-    "husky": "^7.0.2",
-    "identity-obj-proxy": "^3.0.0",
-    "jest": "^27.0.6",
-    "prettier": "^2.3.0"
-  },
-  "engines": {
-    "node": "14.x"
-  }
+	"name": "learn-starter",
```
Comment (swarajpure) [inline package.json:2]: Why this change?

File: src/components/UI/navbar/index.js [hunk 11]
```diff
@@ -1,50 +1,231 @@
...
+        })
+        .then((responseJson) => {
+          if (!responseJson.incompleteUserDetails) {
+            setIsLoggedIn(true);
+            setUserData({
+              userName: responseJson.username,
+              firstName: responseJson.first_name,
+            });
+          }
+          return window.location.replace('https://my.realdevsquad.com/signup');
```
Comment (swarajpure) [inline src/components/UI/navbar/index.js:58]: @sumitd94 this will no longer keep members site public, do we want this?

File: src/components/UI/navbar/index.js [hunk 12]
```diff
@@ -1,50 +1,231 @@
...
-              href="https://members.realdevsquad.com/"
-              style={{ color: '#87D870' }}
+    <div className={styles.wrapper}>
+      <>
+        <nav className={styles.navBar}>
+          <div
+            className={styles.hamburger}
+            ref={navbarRef}
+            onClick={() => setToggle(!toggle)}
+            onKeyDown={() => setToggle(!toggle)}
```
Comment (swarajpure) [inline src/components/UI/navbar/index.js:76]: Repetition! Let's make the anonymous function into a named function and reference it from here

File: src/components/UI/navbar/index.js [hunk 13]
```diff
@@ -1,50 +1,231 @@
...
+          >
+            <span className={styles.bar} />
+            <span className={styles.bar} />
+            <span className={styles.bar} />
```
Comment (swarajpure) [inline src/components/UI/navbar/index.js:80]: Can we use an icon or something here?

File: src/components/UI/navbar/index.js [hunk 14]
```diff
@@ -1,50 +1,231 @@
...
+                ? `${styles.navBarLogin}`
+                : `${styles.navBarLogin} d-none`
+            }
+          >
+            <Link href={LOGIN_URL}>
+              <a
+                className={
+                  isLoggedIn
+                    ? `${styles.btnLogin} d-none`
+                    : `${styles.btnLogin}`
```
Comment (swarajpure) [inline src/components/UI/navbar/index.js:94]: Repetition again.  How about?
```suggestion 
: className={`${styles.btnLogin}${isLoggedIn ? 'd-none' : ''}`}
```


Also what is 'd-none'? Can we use a better variable name?

File: src/components/UI/navbar/index.js [hunk 15]
```diff
@@ -1,50 +1,231 @@
...
+                    width="15px"
+                  />
+                </button>
+              </a>
+            </Link>
+            <div
+              className={
+                isLoggedIn
+                  ? `${styles.userGreet}`
+                  : `${styles.userGreet} d-none`
```
Comment (swarajpure) [inline src/components/UI/navbar/index.js:113]: Same as above

File: src/components/UI/navbar/index.js [hunk 16]
```diff
@@ -1,50 +1,231 @@
...
+                  />
+                </a>
+              </Link>
+            </div>
+          </div>
+          <ul
+            className={
+              toggle
+                ? `${styles.navBarMenu} ${styles.active}`
+                : styles.navBarMenu
```
Comment (swarajpure) [inline src/components/UI/navbar/index.js:140]: Same as above

File: src/components/UI/navbar/index.js [hunk 17]
```diff
@@ -1,50 +1,231 @@
...
+                  src="/images/Real-Dev-Squad@1x.png"
+                  alt="real-dev-squad"
+                />
+              </a>
+            </li>
+            <li>
+              <Link href={PATHS.WELCOME}>Welcome</Link>
+            </li>
+            <li>
+              <Link href={PATHS.EVENTS}>Events</Link>
```
Comment (swarajpure) [inline src/components/UI/navbar/index.js:157]: @rohan09-raj do you have some suggestion here? Familiar scenario?

File: src/components/UI/navbar/navbar.module.scss [hunk 18]
```diff
@@ -1,53 +1,326 @@
...
+  .userGreet {
+    margin: 0px;
   }
 }
+
+// **************************************************************
+
+// .infoNavbar nav ul {
+//   list-style-type: none;
+//   margin: 0;
```
Comment (swarajpure) [inline src/components/UI/navbar/navbar.module.scss:199]: Keeping dead code?

File: src/components/layout/index.js [hunk 19]
```diff
@@ -1,16 +1,16 @@
...
 const Layout = ({ children, title, description }) => {
-  const [showSideDrawer, setShowSideDrawer] = useState(false);
+  // const [showSideDrawer, setShowSideDrawer] = useState(false);
 
-  const sideDrawerToggleHandler = () => {
-    setShowSideDrawer((prevState) => !prevState);
-  };
+  // const sideDrawerToggleHandler = () => {
+  //   setShowSideDrawer((prevState) => !prevState);
+  // };
```
Comment (swarajpure) [inline src/components/layout/index.js:13]: Re: dead code

**Review by swarajpure — CHANGES_REQUESTED** [review]

File: src/components/UI/navbar/index.js (hunk 13 above)
Comment (Kshashank99) [reply src/components/UI/navbar/index.js:80]: Yes we can. But crypto site a dn other site has the same format so i didn't touch it

File: src/components/UI/navbar/index.js (hunk 14 above)
Comment (Kshashank99) [reply src/components/UI/navbar/index.js:94]: That displays none for ul in navbar when we switch to smaller device.

File: src/components/UI/navbar/index.js (hunk 17 above)
Comment (rohan09-raj) [reply src/components/UI/navbar/index.js:157]: Yeah, I have asked him to use an Array of objects here as the repetition of code is there.
@Kshashank99 Can you please do the needful ??

File: src/components/UI/side-drawer/side-drawer.module.scss [hunk 20]
```diff
@@ -16,7 +16,13 @@
   text-align: center;
   padding-top: 20px;
   color: #1d1283;
+  display: flex;
+  flex-direction: row-reverse;
 }
+.logo {
+  margin-right: 2.8125rem;
```
Comment (rohan09-raj) [inline src/components/UI/side-drawer/side-drawer.module.scss:23]: Why are we using the value till this many decimal places ??
Can we please use pixel here ??

File: src/components/layout/index.js [hunk 21]
```diff
@@ -19,8 +19,9 @@ const Layout = ({ children, title, description }) => {
         <meta name="description" content={description} />
         <meta name="viewport" content="width=device-width, initial-scale=1.0" />
       </Head>
-      <Navbar drawerToggleClicked={sideDrawerToggleHandler} />
-      <SideDrawer open={showSideDrawer} close={sideDrawerToggleHandler} />
+      <Navbar />
+      {/* <Navbar drawerToggleClicked={sideDrawerToggleHandler} /> */}
+      {/* <SideDrawer open={showSideDrawer} close={sideDrawerToggleHandler} /> */}
```
Comment (rohan09-raj) [inline src/components/layout/index.js:24]: Please remove commented code

**Review by rohan09-raj — COMMENTED** [review]
@Kshashank99  Can you please remove all the previous attached images in conversation and only upload the current updated UI screenshot in the description. It is bit confusing for us to understand which is the latest one. 
Upload the following screenshots :
- Desktop View
- Mobile view (navbar menu closed)
- Mobile view (navbar menu open)

**Comment by Kshashank99** [comment]
@rohan09-raj @sumitd94 @swarajpure Please review the PR #265 :
## Before: 
### Desktop view 
image
### Mobile view
image
## After:
### Desktop view 
image
### Mobile view
image

File: src/components/UI/navbar/index.js (hunk 14 above)
Comment (swarajpure) [reply src/components/UI/navbar/index.js:97]: Did you check the suggestion?

File: src/components/UI/side-drawer/side-drawer.module.scss (hunk 20 above)
Comment (Kshashank99) [reply src/components/UI/side-drawer/side-drawer.module.scss]: Deleted sidebar component

File: src/components/UI/modal/index.jsx [hunk 22]
```diff
@@ -10,7 +10,7 @@ const Modal = ({ show, closeModal, children, style }) => (
       className={styles.modal}
       style={{
         ...style,
-        transform: show ? 'translateY(0)' : 'translateY(-100vh)',
+        transform: show ? 'translateY(0)' : 'translateY(-10vh)',
```
[the remark this answers is not in the document]
Comment (Kshashank99) [reply src/components/UI/modal/index.jsx:13]: Yes will be deleting this component

File: src/components/UI/backdrop/backdrop.module.scss [hunk 23]
```diff
@@ -0,0 +1,9 @@
+.Backdrop {
```
Comment (rohan09-raj) [inline src/components/UI/backdrop/backdrop.module.scss:1]: Can we please make this starting with small letter ??

File: src/components/UI/backdrop/index.js [hunk 24]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
```
Comment (rohan09-raj) [inline src/components/UI/backdrop/index.js:2]: Re : remove commented code if not in use

File: src/components/UI/navbar/navbar.module.scss [hunk 25]
```diff
@@ -107,8 +108,9 @@
 
 .githubLogo {
   margin-left: 4px;
-  background-color: white;
+  // background-color: white;
```
Comment (rohan09-raj) [inline src/components/UI/navbar/navbar.module.scss:111]: Re : remove commented code

File: src/components/UI/navbar/navbar.module.scss [hunk 26]
```diff
@@ -133,30 +135,28 @@
...
   }
   .hamburger {
     height: 73px;
     width: 73px;
-    padding: 20px;
+    padding: 18px;
     display: inline-block;
     cursor: pointer;
-    font-size: 30px;
+    // font-size: 30px;
```
Comment (rohan09-raj) [inline src/components/UI/navbar/navbar.module.scss:155]: Re: remove commented code

File: src/components/UI/navbar/navbar.module.scss [hunk 27]
```diff
@@ -133,30 +135,28 @@
...
     color: white;
-
-    .icon {
-      margin-top: 4px;
-      padding-left: 2px;
-    }
+  }
```
Comment (rohan09-raj) [inline src/components/UI/navbar/navbar.module.scss:157]: Please add a line space after this

**Review by rohan09-raj — CHANGES_REQUESTED** [review]

File: src/components/UI/navbar/index.js [hunk 28]
```diff
@@ -55,6 +55,9 @@ const Navbar = () => {
             setUserData({
               userName: responseJson.username,
               firstName: responseJson.first_name,
+              profilePicture: responseJson.picture
+                ? responseJson.picture.url
+                : DEFAULT_AVATAR,
             });
           }
           return window.location.replace('https://my.realdevsquad.com/signup');
```
[the remark this answers is not in the document]
Comment (Kshashank99) [reply src/components/UI/navbar/index.js:63]: I didn't get the workflow. Does it mean if the profilePicture is Default_AVATAR, it will get open in a new tab. But why open it in a new window in the first place until the user does not click on it.

[the remark this answers is not in the document]
Comment (rohan09-raj) [reply src/components/UI/navbar/index.js:63]: No I am asking to modify that return line 63. which replaces the current URL/ location. We can open in new tab because what if user is not logged in and this will redirect to `/signup` flow and ask the user to login. Then, we lost the crypto page because the site is opening in the same tab. 

What's your take on this ??

[the remark this answers is not in the document]
Comment (Kshashank99) [reply src/components/UI/navbar/index.js:63]: Yeah got your point. well its the member site not crypto😅

[the remark this answers is not in the document]
Comment (rohan09-raj) [reply src/components/UI/navbar/index.js:63]: Ahh 😂 sorry, I was reviewing the Shubham's PR for improvement of navbar in crypto site, that's why mistook it.

File: src/components/UI/navbar/index.js [hunk 29]
```diff
@@ -55,6 +55,9 @@ const Navbar = () => {
             setUserData({
               userName: responseJson.username,
               firstName: responseJson.first_name,
+              profilePicture: responseJson.picture
+                ? responseJson.picture.url
```
[the remark this answers is not in the document]
Comment (Kshashank99) [reply src/components/UI/navbar/index.js:59]:
```suggestion
                ? responseJson.picture.url || DEFAULT_AVATAR
```
I think this will work

File: src/components/UI/navbar/index.js [hunk 30]
```diff
@@ -155,7 +144,10 @@ const Navbar = () => {
             </li>
             {NAVMENU.map((nav) => {
               return (
-                <li className={nav.name === 'Home' ? `${styles.Homes}` : null}>
+                <li
+                  className={nav.name === 'Home' ? `${styles.homes}` : null}
+                  key
```
[the remark this answers is not in the document]
Comment (Kshashank99) [reply src/components/UI/navbar/index.js:149]: Added Id

File: src/components/UI/navbar/navbar.module.scss (hunk 25 above)
Comment (Kshashank99) [reply src/components/UI/navbar/navbar.module.scss:111]: Done

File: src/components/UI/navbar/navbar.module.scss [hunk 31]
```diff
@@ -107,8 +108,9 @@
 
 .githubLogo {
   margin-left: 4px;
-  background-color: white;
+  // background-color: white;
   border-radius: 50%;
+  height : 15px; width : 15px;
```
[the remark this answers is not in the document]
Comment (Kshashank99) [reply src/components/UI/navbar/navbar.module.scss:113]: Done

File: src/components/UI/navbar/navbar.module.scss (hunk 26 above)
Comment (Kshashank99) [reply src/components/UI/navbar/navbar.module.scss:155]: Done

File: src/components/UI/navbar/index.js (hunk 29 above)
[the remark this answers is not in the document]
Comment (rohan09-raj) [reply src/components/UI/navbar/index.js:59]: I think this will only work when the request fails and there is no response object, what do you think ??

[the remark this answers is not in the document]
Comment (Kshashank99) [reply src/components/UI/navbar/index.js:59]: If the request fails then this code will never get executed because it won't enter the if block right?

[the remark this answers is not in the document]
Comment (rohan09-raj) [reply src/components/UI/navbar/index.js:59]: Ok not exactly failure of request, but it may happen that the data from backend changes somehow and there is no URL property. So just to be safe from breaking of site, we use optional chaining.

[the remark this answers is not in the document]
Comment (Kshashank99) [reply src/components/UI/navbar/index.js:59]: Ok I need to read about optional chaining first.

**Review by rohan09-raj — DISMISSED** [review]
Looks good to me 💯 , thanks for being patient and actively addressing comments. Really appreciate the efforts. Approved from my side
